makefiles/pseudomodules.inc.mk

```
@@ -122,9 +122,7 @@ PSEUDOMODULES += ipv6
 PSEUDOMODULES += l2filter_blacklist
 PSEUDOMODULES += l2filter_whitelist
 PSEUDOMODULES += libstdcpp
-PSEUDOMODULES += log
-PSEUDOMODULES += log_printfnoformat
-PSEUDOMODULES += log_color
+PSEUDOMODULES += log_%
```

Why reduce specificity?
```suggestion
PSEUDOMODULES += log
PSEUDOMODULES += log_printfnoformat
PSEUDOMODULES += log_color
```

---

It seemed overspecific (i had to remove `PSEUDOMODULES += log` since there are now variants that have build results)
so i made it less specific (as it is for other pseudo modules)

---

if log is a pseudo module the build result isn't linked

---

So how does a user find out what `log_...` pseudo-modules there are if you remove the only bit of documentation?

---

> It seemed overspecific (i had to remove `PSEUDOMODULES += log` since there are now variants that have build results)

Sorry my mistake. I realized that, but for got to remove the line. Fixed in the suggestion.

---

> So how does a user find out what `log_...` pseudo-modules there are if you remove the only bit of documentation?

i never saw (the quiet hidden pseudomodules.inc.mk) as documentation, and there seem are a lot on % in it (e.g.: prng_%)

https://doc.riot-os.org/group__sys__log__color.html seems like documentation of that feature and 
https://doc.riot-os.org/group__sys__log__printfnoformat.html for the other one

---

> and there seem are a lot on % in it (e.g.: prng_%)

And I dislike them as much as what you are trying to do here. I  was and always will be against wildcards to define pseudomodules. I don't see any reason for them except lazyness. Let's be good scouts and not make the situation worse than it already is, just because there is precedence for bad behavior.

As for documentation: This file is actually rendered by Doxygen: https://doc.riot-os.org/group__pseudomodules.html...

---

https://output.circle-artifacts.com/output/job/a338c64b-59b4-4911-907d-afcce6e4bdf8/artifacts/0/doc/group__pseudomodules.html

---

If you can explain to me, why the change to wildcards is so important I am inclined to remove my change request regarding this. Because I still don't really see a valid reason.

> https://output.circle-artifacts.com/output/job/a338c64b-59b4-4911-907d-afcce6e4bdf8/artifacts/0/doc/group__pseudomodules.html

Why do they use a different formatting than the rest of the listed pseudo-modules there?

---

> Why do they use a different formatting than the rest of the listed pseudo-modules there?

While most pseudomodules aren't even listed the one that are use the defgroup title in entirely different way than the rest of the documentation
I think: the title of a common defgroup for a module should be as i just wrote the one for log_... `<modulename>: <title>` this would also help the list in  https://doc.riot-os.org/group__sys.html

eg.: from the syslist i know the is a "Lightweight Morse encoder" - what is its module to use?
changing this to "umorse: Lightweight Morse encoder" would help a lot

atm for many sys-modules the title is missused as a brief:
CongURE and fido2  are nice exemptions

---

There is https://github.com/RIOT-OS/RIOT/issues/7094. `<modulename>: <title>` would be one way to fix this. However, IMHO this should be done all in one go.